retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/RetrofitMeterIdPrefixFunctionBuilder.java

```diff
@@ -51,10 +54,23 @@ public RetrofitMeterIdPrefixFunctionBuilder withServiceTag(String serviceTagName
         return this;
     }
 
+    /**
+     * Adds a service that will allow to record actual path for each method in addition to method name.
+     *
+     * @param serviceClass the class defining client with Retrofit.
+     */
+    public RetrofitMeterIdPrefixFunctionBuilder withServiceClass(Class serviceClass) {
+        this.serviceClass = requireNonNull(serviceClass, "serviceClass");
+        return this;
+    }
+
     /**
      * Returns a newly created {@link RetrofitMeterIdPrefixFunction} with the properties specified so far.
      */
-    public RetrofitMeterIdPrefixFunction build() {
-        return new RetrofitMeterIdPrefixFunction(name, serviceTagName, defaultServiceName);
+    public MeterIdPrefixFunction build() {
```

Comment:
We can continue returning `RetrofitMeterIdPrefixFunction` if we make `RetrofitClassAwareMeterIdPrefixFunction` extend `RetrofitMeterIdPrefixFunction`. Let's make `RetrofitMeterIdPrefixFunction` non-final with package-private constructor and let `RetrofitClassAwareMeterIdPrefixFunction` extend it.